addons/web/static/src/core/file_viewer/file_viewer.js

```diff
@@ -207,14 +277,24 @@ export class FileViewer extends Component {
     }
 
     updateZoomerStyle() {
-        const tx =
-            this.imageRef.el.offsetWidth * this.state.scale > this.zoomerRef.el.offsetWidth
-                ? this.translate.x + this.translate.dx
-                : 0;
-        const ty =
-            this.imageRef.el.offsetHeight * this.state.scale > this.zoomerRef.el.offsetHeight
-                ? this.translate.y + this.translate.dy
-                : 0;
+        const isImageRotated = this.state.angle % 180 !== 0;
+        const { offsetWidth, offsetHeight } = this.imageRef.el;
+        const imageWidth = (isImageRotated ? offsetHeight : offsetWidth) * this.state.scale;
+        const imageHeight = (isImageRotated ? offsetWidth : offsetHeight) * this.state.scale;
+        const diffX = imageWidth - this.zoomerRef.el.offsetWidth + 20;
+        const diffY = imageHeight - this.zoomerRef.el.offsetHeight + 120;
+        let tx = diffX > 0 ? this.translate.x + this.translate.dx : 0;
+        let ty = diffY > 0 ? this.translate.y + this.translate.dy : 0;
```

Comment:
What are these magic numbers `20` and `120`?
If they refer to height and width on something on UI, should use `ref.el.clientWidth` or something alike.